### PR TITLE
New formula: Zeal 0.1.1

### PR DESCRIPTION
**Pull request by benjaminfrank**
Offline documentation browser inspired by Dash.

### CODE REVIEW

File: Library/Formula/zeal.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  desc "Zeal is a simple offline documentation browser inspired by Dash."
+  homepage "http://zealdocs.org/"
+  url "https://github.com/zealdocs/zeal/archive/v0.1.1.tar.gz"
+  sha256 "f4e959f9bc66a6e350ee8a33d34695379633432d103db9776c0c7a76cbc5a9d6"
+  head "https://github.com/zealdocs/zeal.git"
+
+  depends_on "qt5"
+  depends_on "libarchive"
+
+  patch :DATA
```
Comment (DomT4) [inline Library/Formula/zeal.rb]: Can you report this upstream if it hasn't been already and link to that bug report? Thanks!

File: Library/Formula/zeal.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    system "#{qt5.installed_prefix}/bin/qmake"
+    system "make"
+    system "#{qt5.installed_prefix}/bin/macdeployqt", "bin/Zeal.app"
+    prefix.install "bin/Zeal.app"
+    (bin/"zeal").write("#! /bin/sh\n#{prefix}/Zeal.app/Contents/MacOS/Zeal \"$@\"\n")
+  end
+
+  test do
+    system "zeal", "-h"
```
Comment (DomT4) [inline Library/Formula/zeal.rb]: Can a test be added to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Library/Formula/zeal.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  head "https://github.com/zealdocs/zeal.git"
+
+  depends_on "qt5"
+  depends_on "libarchive"
+
+  patch :DATA
+
+  def install
+    qt5 = Formula["qt5"]
+    ENV.append "CFLAGS", "-I#{qt5.installed_prefix}/include -I#{HOMEBREW_PREFIX}/opt/libarchive/include"
```
Comment (DomT4) [inline Library/Formula/zeal.rb]: `-I#{Formula["qt5"].opt_include} -I#{Formula["libarchive"].opt_include}` here. The `qt5` one should be being passed automatically. Does it not pick up on that?

File: Library/Formula/zeal.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    ENV.append "CXXFLAGS", "-I#{qt5.installed_prefix}/include -I#{HOMEBREW_PREFIX}/opt/libarchive/include"
```
Comment (DomT4) [inline Library/Formula/zeal.rb]: Same thing here and for `LDFLAGS` as well.

File: Library/Formula/zeal.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    ENV.append "LDFLAGS", "-L#{qt5.installed_prefix}/lib -L#{HOMEBREW_PREFIX}/opt/libarchive/lib -larchive"
+
+    system "#{qt5.installed_prefix}/bin/qmake"
```
Comment (DomT4) [inline Library/Formula/zeal.rb]: This can just be `system "qmake"`.

File: Library/Formula/zeal.rb [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    system "make"
+    system "#{qt5.installed_prefix}/bin/macdeployqt", "bin/Zeal.app"
```
Comment (DomT4) [inline Library/Formula/zeal.rb]: Same here. `system "macdeployqt"`

File: Library/Formula/zeal.rb [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+--- a/src/src.pro
++++ b/src/src.pro
+@@ -78,3 +78,8 @@ MOC_DIR = $$BUILD_ROOT/.moc
+ OBJECTS_DIR = $$BUILD_ROOT/.obj
+ RCC_DIR = $$BUILD_ROOT/.rcc
+ UI_DIR = $$BUILD_ROOT/.ui
++QMAKE_CFLAGS += $$(CFLAGS)
++QMAKE_CXXFLAGS += $$(CXXFLAGS)
++QMAKE_LFLAGS += $$(LDFLAGS)
++macx:INCLUDEPATH += /usr/local/opt/libarchive/include
```
Comment (DomT4) [inline Library/Formula/zeal.rb]: This won't always be true. Homebrew can be placed anywhere - `/usr/local` is just the default.

File: Library/Formula/zeal.rb [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
++macx:DEPENDPATH += /usr/local/opt/libarchive/include
```
Comment (DomT4) [inline Library/Formula/zeal.rb]: Same here.

**Comment by benjaminfrank** [comment]
I just found https://github.com/zealdocs/zeal/issues/24
So maybe I should clarify if adding zeal to homebrew creates any issue before you merge. Thanks.

**Comment by benjaminfrank** [comment]
While I tried to fix the mentioned issues with this formula, I's still like to wait for the zeal/dash guys to give their ok.

**Comment by benjaminfrank** [comment]
As agreed with the maintainers of Zeal/Dash I will close this PR and would like to ask you not to include my formula into homebrew. Thanks, and sorry for the overhead.

**Comment by mikemcquaid** [comment]
Thanks for checking.

**Pull request closed without merging**